docs/guide/migration-from-0.6-to-1.0.md

```diff
@@ -0,0 +1,199 @@
+# Migrating from 0.6 to 1.0
+
+To upgrade your HyperFormula version from 0.6.x to 1.0.x, follow this guide.
+
+## Step 1: Change your license key
+
+Change your HyperFormula license key.
+
+For more details on HyperFormula license keys, go [here](license-key.md).
+
+### Open-source license
+
+If you use the open-source version of HyperFormula, in your configuration options, pass the `gpl-v3` string instead of the `agpl-v3` string:
+
+Before:
+```js
+const options = {
+  licenseKey: 'agpl-v3',
+}
+```
+
+After:
+```js
+const options = {
+  // use `gpl-v3` instead of `agpl-v3`
+  licenseKey: 'gpl-v3',
+}
+```
+
+### Free non-commercial license
+
+If you use the free non-commercial license, switch to the GPLv3 license or purchase a commercial license.
+
+### Commercial license
+
+If you use a commercial license, you don't need to make any changes.
+
+## Step 2: Change `sheetName` to `sheetId`
+
+Most sheet-related methods now take the `sheetID` number parameter instead of the `sheetName` string parameter.
+
+For example, use the `clearSheet()` method in this way:
+
+Before:
+```js
+const hfInstance = HyperFormula.buildFromSheets({
+ MySheet1: [ ['=SUM(MySheet2!A1:A2)'] ],
+ MySheet2: [ ['10'] ],
+});
+
+const changes = hfInstance.clearSheet('MySheet2');
+```
+
+After:
+```js
+const hfInstance = HyperFormula.buildFromSheets({
+ MySheet1: [ ['=SUM(MySheet2!A1:A2)'] ],
+ MySheet2: [ ['10'] ],
+});
+
+// use `sheetId` instead of `sheetName`
+const changes = hfInstance.clearSheet(0);
+```
+
+The only methods still accepting the `sheetName` string parameter are now:
+- `addSheet()`
+- `isItPossibleToAddSheet()`
+- `doesSheetExist()`
+- `getSheetId()`
```

Comment:
Maybe some explanation why those are not affected? Might be obvious, they need the name to work correctly.
- addSheet -> name that sheet will get, ID is generated 
- isItPossibleToAddSheet -> WITH that name
- doesSheetExist -> with that name
- getSheetId -> doh, you can't get an id from the id

Reply:
@wojciechczerniak - done: 21f936a